src/commcare_cloud/fab/operations/release.py

```diff
@@ -47,11 +47,11 @@ def update(git_tag, subdir=None, code_repo=None, deploy_key=None):
             sudo('git remote prune origin')
             # this can get into a state where running it once fails
             # but primes it to succeed the next time it runs
-            sudo('git fetch origin --tags -q || git fetch origin --tags -q')
+            sudo(f'git fetch origin {env.code_branch} --tags -q || git fetch origin {env.code_branch} --tags -q')
             sudo('git checkout {}'.format(git_tag))
             sudo('git reset --hard {}'.format(git_tag))
             sudo('git submodule sync')
-            sudo('git submodule update --init --recursive -q')
+            sudo('git submodule update --init --recursive --single-branch -q')
```

Comment:
Has this been tested and confirmed to avoid fetching ancillary branches on a submodule where:

- parent repo existed (was cloned) prior to _this PR_, and
- the submodule existed (was cloned into the the parent repo) prior to _this PR_

I'm interested in this specific scenario because my understanding of `--single-branch` is that it only has effect on a `git clone` operation (which is only ever performed once per repo; naturally). In my testing, it appears that the _only_ thing that `--single-branch` does is configure the `remote.<name>.fetch` setting on a repo to be restrictive (i.e. only fetch the default remote branch) _at clone time_.

I think this means that any repos that were cloned _prior to this PR_ will continue to indiscriminately fetch _all_ branches, even after this PR is "live", which means that this PR won't help any HQ repos (parent or submodules) until they are either "recycled" (by decommission, manual deletion, etc) or a manual correction with something like:

```shell
git config remote.origin.fetch '+refs/heads/master:refs/remotes/origin/master'
git submodule foreach "git config remote.origin.fetch '+refs/heads/master:refs/remotes/origin/master'"
```